lenskit/algorithms/basic.py

```diff
@@ -538,3 +292,18 @@ def recommend(self, user, n=None, candidates=None, ratings=None):
 
     def __str__(self):
         return 'Random'
+
+
+class KnownRating(Predictor):
+    """
+    The known rating algorithm memorizes ratings provided in the fit method.
+    """
+
+    def fit(self, ratings, **kwargs):
+        self.ratings = ratings
+        return self
+
+    def predict_for_user(self, user, items, ratings=None):
+        uscores = self.ratings[self.ratings.user == user]
```

Comment:
If we set up `ratings` to have an index in `fit`, then this can use an index for better performance on large data sets.